synapse/appservice/api.py

```diff
@@ -190,10 +197,12 @@ async def query_3pe(
         assert service.hs_token is not None
 
         try:
-            args: Mapping[Any, Any] = {
-                **fields,
-                b"access_token": service.hs_token,
-            }
+            args: Mapping[Any, Any] = fields
```

Comment:
Can we remove the `Mapping[Any, Any]`? Curious what mypy error that gives.

Reply:
This is the mypy error:
```
synapse/appservice/api.py:202: error: Dict entry 0 has incompatible type "bytes": "str"; expected "bytes": "List[bytes]"
```
It seems upset with the re-assignment of `args`.

Reply:
I think you can instead do:

```python
args = {
    **fields,
    b"access_token": [service.hs_token],
}
```

I'd really rather not add more `Any` if possible!

Reply:
Switching it to your suggestion results in this error:
```
synapse/appservice/api.py:199: error: Argument 1 to "update" of "MutableMapping" has incompatible type "Dict[bytes, List[bytes]]"; expected "SupportsKeysAndGetItem[bytes, List[str]]"  [arg-type]

```
I don't suppose we can't just use `args: Mapping = fields` (which works) because that's just an implicit `Any`?

Reply:
Does `args: Mapping[bytes, Union[bytes, List[str]]` work? 😆 (You would need to switch back `service.hs_token` back to not a list or do `Union[List[bytes], List[str]]`)